src/components/tikzpictureview.ts

```diff
@@ -0,0 +1,361 @@
+                    continue
+                } else if (tikzPicture.range.start.isAfter(change.range.end)) {
+                    // change before tikzpicture
+                    tikzPicture.range = new vscode.Range(
+                        tikzPicture.range.start.translate(lineDelta, 0),
+                        tikzPicture.range.end.translate(lineDelta, 0)
+                    )
+                } else if (
+                    change.range.end.isAfter(tikzPicture.range.start) &&
+                    change.range.end.line - lineDelta < tikzPicture.range.start.line
+                ) {
+                    // tikzpicture removed
+                    tikzPictures.splice(tikzPictures.indexOf(tikzPicture), 1)
+                    this.cleanupTikzPicture(tikzPicture)
+                } else {
+                    // tikzpicture modified
+                    let startLocation: vscode.Position | null = null
+                    if (change.range.start.line <= tikzPicture.range.start.line) {
+                        const startLine = document.lineAt(tikzPicture.range.start.line)
+                        const tikzPictureStartIndex = startLine.text.indexOf('\\begin{tikzpicture}')
+                        if (tikzPictureStartIndex !== -1) {
+                            startLocation = tikzPicture.range.start.translate(
+                                0,
+                                tikzPictureStartIndex - tikzPicture.range.start.character
+                            )
+                        } else {
+                            const startRegex = /\\begin{tikzpicture}/
+                            let startMatch: RegExpMatchArray | null = null
+                            let lineNo = change.range.start.line - 1
+                            do {
+                                startMatch = document.lineAt(++lineNo).text.match(startRegex)
+                            } while (!startMatch && lineNo <= tikzPicture.range.end.line)
+
+                            if (startMatch) {
+                                // @ts-ignore
+                                startLocation = new vscode.Position(lineNo, startMatch.index)
+                            }
+                        }
+                    }
+
+                    let endLocation: vscode.Position | null = null
+                    if (change.range.end.line >= tikzPicture.range.end.line) {
+                        // things can be a bit funny so we'll just look for the matchin \end{tikzpicture}
+                        const endRegex = /\\end{tikzpicture}/
+                        let endMatch: RegExpMatchArray | null = null
+                        let lineNo = tikzPicture.range.start.line - 1
+                        do {
+                            endMatch = document.lineAt(++lineNo).text.match(endRegex)
+                        } while (!endMatch && lineNo <= change.range.end.line)
+
+                        if (endMatch) {
+                            // @ts-ignore
```

Comment:
Don't use @ts-ignore. Fix it.

Reply:
See 86daf0ec53d032b66a5fa7a69bfbd9845ab2cdc2